lib/task_helper.rb

```diff
@@ -0,0 +1,48 @@
+require 'json'
+
+class TaskHelper
+  class Error < RuntimeError
+    attr_reader :kind, :details, :issue_code
+
+    def initialize(msg, kind, details = nil)
+      super(msg)
+      @kind = kind
+      @issue_code = issue_code
+      @details = details || {}
+    end
+
+    def to_h
+      { 'kind' =>  kind,
+        'msg' => message,
+        'details' => details }
+    end
+  end
+
+  def task(params = {})
+    msg = 'The task author must implement the `task` method in the task'
+    error = TaskHelper::Error.new(msg, 'tasklib/not-implemented')
+    STDOUT.print error.to_h.to_json
```

Comment:
This should just raise the error, so that it gets printed by the error handler in `run`.